blueocean-dashboard/src/main/js/redux/actions.js

```diff
@@ -57,7 +57,23 @@ export const actionHandlers = {
         return state.set('currentRuns', null);
     },
     [ACTION_TYPES.SET_CURRENT_RUN_DATA](state, { payload }): State {
-        return state.set('currentRuns', payload);
+        return state.set('currentRuns', payload.map((run) => {
+            if (run._class === 'io.jenkins.blueocean.service.embedded.rest.QueueItemImpl') {
```

Comment:
I am also not sure how it applies to queued items. Capabilities say nothing
about queued items.

It would need to be contorted to describe it in some way.

Ivan the capabilities API is too difficult for his you think?
On Thu, 4 Aug 2016 at 8:20 PM, Ivan Meredith <notifications@github.com>
wrote:

> In blueocean-dashboard/src/main/js/redux/actions.js
> <https://github.com/jenkinsci/blueocean-plugin/pull/389#discussion_r73497156>
> :
>
> > @@ -57,7 +57,23 @@ export const actionHandlers = {
> >          return state.set('currentRuns', null);
> >      },
> >      [ACTION_TYPES.SET_CURRENT_RUN_DATA](state, { payload }): State {
> > -        return state.set('currentRuns', payload);
> > +        return state.set('currentRuns', payload.map((run) => {
> > +            if (run._class === 'io.jenkins.blueocean.service.embedded.rest.QueueItemImpl') {
>
> Actually I have no idea how to use capability api here in an async fashion
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/blueocean-plugin/pull/389/files/272b537e658774701b872f9d0053a45e0c11149d#r73497156>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAA6gAFNLo73TNYMMYwUqgjcvPy5ftv2ks5qcbztgaJpZM4JbQis>
> .
>
-- 

-- 
Regards,

Michael Neale
(twitter: @michaelneale, skype: michael_d_neale)
Cell: +61 423175597
Cofounder & Engineer @ large in CTO office.